jme3-lwjgl3/src/main/java/com/jme3/system/lwjgl/LwjglCanvas.java

```diff
@@ -0,0 +1,372 @@
+                    }
+                    
+                    AffineTransform at = gc.getDefaultTransform();
+                    float sx = (float) at.getScaleX(),
+                          sy = (float) at.getScaleY();
+
+                    int fw = (int) (canvas.getWidth() * sx);
+                    int fh = (int) (canvas.getHeight() * sy);
+
+                    if (fw != framebufferWidth || fh != framebufferHeight) {
+                        framebufferWidth = Math.max(fw, 1);
+                        framebufferHeight = Math.max(fh, 1);
+                        needResize.set(true);
+                    }
+                }
+            }            
+        });
+    }
+    
+    @Override
+    protected void createContext(AppSettings settings)  {
+        super.createContext(settings);
+        if (settings.getBitsPerPixel() == 24) {
+            glData.redSize = 8;
+            glData.greenSize = 8;
+            glData.blueSize = 8;            
+        } else if (settings.getBitsPerPixel() == 16) {
+            glData.redSize = 5;
+            glData.greenSize = 6;
+            glData.blueSize = 5;            
+        }
+        
+        glData.depthSize = settings.getBitsPerPixel();
+        glData.alphaSize = settings.getAlphaBits();
+        glData.sRGB = settings.isGammaCorrection();
+        
+        glData.depthSize = settings.getDepthBits();
+        glData.stencilSize = settings.getStencilBits();
+        glData.samples = settings.getSamples();
+        glData.stereo = settings.useStereo3D();  
+       
+        glData.debug = settings.isGraphicsDebug();
+        glData.profile = GLData.Profile.CORE;
+        glData.api = GLData.API.GL;
+    }
+    
+    @Override
+    public Canvas getCanvas() {
+        return canvas;
+    }
+
+    @Override
+    protected void showWindow() {
+    }
+
+    @Override
+    protected void setWindowIcon(final AppSettings settings) {
+    }
+
+    @Override
+    public void setTitle(String title) {
+    }
+
+    @Override
+    public KeyInput getKeyInput() {
+        if (keyInput == null) {
+            keyInput = new AwtKeyInput();
+            keyInput.setInputSource(canvas);
+        }
+        return keyInput;
+    }
+
+    @Override
+    public MouseInput getMouseInput() {
+        if (mouseInput == null) {
+            mouseInput = new AwtMouseInput();
+            mouseInput.setInputSource(canvas);
+        }
+        return mouseInput;
+    }
+
+    public boolean checkVisibilityState() {
+        if (!hasNativePeer.get()) {
+            synchronized (lock) {
+                canvas.doDisposeCanvas();
+            }
+            return false;
+        }
+
+        boolean currentShowing = canvas.isShowing();
+        showing.set(currentShowing);
+        return currentShowing;
+    }
+
+
+    @Override
+    protected void destroyContext() {
+         synchronized (lock) {
+            canvas.deleteContext();
+        }
+        //// request the cleanup
```

Comment:
Code in comments generally should not be. Is this something that is incomplete?

Reply:
It shouldn't be there, I forgot to delete it. Sorry for causing confusion...